firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteRemoteDocumentCache.java

```diff
@@ -171,7 +182,10 @@ public Map<DocumentKey, MutableDocument> getAll(
    * Returns the next {@code count} documents from the provided collections, ordered by read time.
    */
   private Map<DocumentKey, MutableDocument> getAll(
-      List<ResourcePath> collections, IndexOffset offset, int count) {
+      List<ResourcePath> collections,
+      IndexOffset offset,
+      int count,
+      @Nullable DocumentFilter filter) {
```

Comment:
You can remove the `DocumentFilter` interface and use `Function`

e.g.

```java
private Map<DocumentKey, MutableDocument> getAll(
      List<ResourcePath> collections,
      IndexOffset offset,
      int count,
      @Nullable Function<MutableDocument> documentFilter) {
      // ...
}
  private void processRowInBackground(
      BackgroundQueue backgroundQueue,
      Map<DocumentKey, MutableDocument> results,
      Cursor row,
      @Nullable Function<MutableDocument> documentFilter) {
      // ...
      // Use documentFilter.apply(document)
}
  @Override
  public Map<DocumentKey, MutableDocument> getDocumentsMatchingQuery(
      Query query, IndexOffset offset, @Nonnull Set<DocumentKey> mutatedKeys) {
    return getAll(
        Collections.singletonList(query.getPath()),
        offset,
        Integer.MAX_VALUE,
        (MutableDocument doc) -> query.matches(doc) || mutatedKeys.contains(doc.getKey()));
  }
```

Reply:
Done.